include/aws_iot_mqtt_client.h

```diff
@@ -312,112 +312,181 @@ struct _Client {
+ * or @ref mqtt_function_yield.
  */
+/* @[declare_mqtt_set_connect_params] */
 IoT_Error_t aws_iot_mqtt_set_connect_params(AWS_IoT_Client *pClient, IoT_Client_Connect_Params *pNewConnectParams);
+/* @[declare_mqtt_set_connect_params] */
 
 /**
- * @brief Is the MQTT client currently connected?
+ * @brief Determine if the MQTT client context currently connected to a server.
+ *
+ * This function checks the internal state of the MQTT client context to determine
+ * if it is currently connected to the server.
  *
- * Called to determine if the MQTT client is currently connected.  Used to support logic
- * in the device application around reconnecting and managing offline state.
+ * @param[in] pClient MQTT client context
  *
- * @param pClient Reference to the IoT Client
+ * @return true if connected; false otherwise.
  *
- * @return true = connected, false = not currently connected
+ * @warning Application code should not rely on this function's return value.
+ * It is retrieved solely from the internal state of the client, and therefore
+ * does not actually check the network or server's status. Additionally,
+ * the connection status may change between the time this function is called
+ * and the application checks its return value.
  */
+/* @[declare_mqtt_is_client_connected] */
 bool aws_iot_mqtt_is_client_connected(AWS_IoT_Client *pClient);
+/* @[declare_mqtt_is_client_connected] */
 
 /**
- * @brief Get the current state of the client
+ * @brief Get the current state of the MQTT client context.
  *
- * Called to get the current state of the client
+ * @param[in] pClient MQTT client context
  *
- * @param pClient Reference to the IoT Client
+ * @return The state of the MQTT client context at the time of the function call.
  *
- * @return ClientState value equal to the current state of the client
+ * @warning Application code should not rely on this function's return value.
+ * The client's state may change between the time this function is called
+ * and the application checks its return value.
  */
+/* @[declare_mqtt_get_client_state] */
 ClientState aws_iot_mqtt_get_client_state(AWS_IoT_Client *pClient);
+/* @[declare_mqtt_get_client_state] */
 
 /**
- * @brief Is the MQTT client set to reconnect automatically?
+ * @brief Determine if auto-reconnect is enabled for an MQTT client context.
  *
- * Called to determine if the MQTT client is set to reconnect automatically.
- * Used to support logic in the device application around reconnecting
+ * @param[in] pClient MQTT client context
  *
- * @param pClient Reference to the IoT Client
- *
- * @return true = enabled, false = disabled
+ * @return true if auto-reconnect is enabled; false otherwise.
  */
+/* @[declare_mqtt_is_autoreconnect_enabled] */
 bool aws_iot_is_autoreconnect_enabled(AWS_IoT_Client *pClient);
+/* @[declare_mqtt_is_autoreconnect_enabled] */
 
 /**
- * @brief Set the IoT Client disconnect handler
+ * @brief Reset the disconnect handler of an initialized MQTT client context.
```

Comment:
Should there be a warning here non-thread safety in case of concurrent disconnection and update of disconnect handler?

Reply:
I think the warning about not calling concurrently with yield is sufficient.  The disconnect handler will only be called through yield.